frontend/src/frontend/code_generator/blocks.py

```diff
@@ -128,20 +129,67 @@ def next_target_sequence(self, length: int = 16) -> List[int]:
         return paths
 
 
+class CacheLine:
+
+    def __init__(self, size: int = 64) -> None:
+        self.size: int = size
+        self.c_variable: str = 'CACHELINE_SIZE'
+
+
+class TargetType(Enum):
+    UNKNOWN = cfg_pb2.CodePrefetchInst.UNKNOWN
+    FUNCTION = cfg_pb2.CodePrefetchInst.FUNCTION
+    CODE_BLOCK = cfg_pb2.CodePrefetchInst.CODE_BLOCK
+
+
+class CodePrefetchInst:
+    """Prefetch instruction indication."""
+
+    def __init__(self, target_type: cfg_pb2.CodePrefetchInst.TargetTypeValue,
+                 target_name: int, degree: int) -> None:
+        self.type: TargetType = TargetType(target_type)
+        self.target_name: int = target_name
+        self.degree: int = degree
+
+    @classmethod
+    def from_proto(cls,
+                   proto_cpi: cfg_pb2.CodePrefetchInst) -> CodePrefetchInst:
+        return cls(proto_cpi.type, proto_cpi.target_id, proto_cpi.degree)
+
+
 class CodeBlockBody:
     """Contains instructions."""
 
-    def __init__(self, name: int, instructions: str) -> None:
+    def __init__(self,
+                 name: int,
+                 instructions: Optional[str] = None,
+                 prefetch_inst: Optional[CodePrefetchInst] = None) -> None:
         self.name: int = name
-        self.instructions: str = instructions
+        self._instructions: Optional[str] = instructions
+        self._prefetch_inst: Optional[CodePrefetchInst] = prefetch_inst
 
     def __str__(self) -> str:
         return (f'CodeBlockBody(name: {self.name}, '
-                f'instructions: {self.instructions})')
+                f'instructions: {self._instructions})')
```

Comment:
Shouldn't this be conditional on whether this body is a set of instructions or a prefetch hint?

Reply:
Yes. Changed in next patch.